### PR TITLE
gateway-api: appProtocol support (GEP-1911)

### PR DESCRIPTION
**Pull request by rauanmayemir**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
docs good

**Comment by rauanmayemir** [comment]
@sayboras 🥹

**Comment by sayboras** [comment]
Thanks for the ping, I have added myself as reviewer.

**Comment by sayboras** [comment]
From a quick look, the changes are looking good to me. Will take a closer look later.

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (sayboras) [inline install/kubernetes/cilium/values.yaml.tmpl:801]: Is there any drawback to have this enabled by default?

Comment (rauanmayemir) [reply install/kubernetes/cilium/values.yaml.tmpl:801]: @sayboras Yes, more on this here: #30452. Basically, if we enable this by default and someone depended on their downstream to choose http protocol, it will break them.

Comment (rauanmayemir) [reply install/kubernetes/cilium/values.yaml.tmpl:801]: @youngnick suggested to put it behind a flag as it's a breaking change.

Comment (sayboras) [reply install/kubernetes/cilium/values.yaml.tmpl:801]: thanks for the pointer

**Comment by rauanmayemir** [comment]
@sayboras `ci-ipsec-upgrade` failure doesn't look like my doing, do I need to check it?

**Review by sayboras — APPROVED** [review]
Thanks and LGTM ✅

**Review by squeed — APPROVED** [review]

**Comment by sayboras** [comment]
Seems like we just need to wait for ack from other codeowner, so you don't need to rebase with main branch (just to avoid re-run the tests).

**Comment by rauanmayemir** [comment]
@sayboras I occasionally rebase to make sure there are no conflicts, will wait for the ack from the codeowner.

**Comment by sayboras** [comment]
> I occasionally rebase to make sure there are no conflicts, will wait for the ack from the codeowner.

Understood and thanks, I normally just rely on github UI to flag if there is any conflict with main branch.

File: operator/pkg/model/translation/cec_translator.go
Comment (rauanmayemir) [inline operator/pkg/model/translation/cec_translator.go:328]: @sayboras I rebased to fix the conflicts and also slightly refactored the code because ifs were becoming a mess.

**Review by youngnick — APPROVED** [review]

**Pull request merged**
